File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
import java.util.*;

class Solution {
    boolean solution(String s) {
        boolean answer = true;

        Deque<Character> stack = new ArrayDeque<>();
        
        for (char p : s.toCharArray()) {
            if (p == '(') {
                stack.push(p);
            } else {
                if(stack.isEmpty()) {
                    answer = false;
                    break;
                } else {
                    char top = stack.pop();
                    if (top == ')') {
                        answer = false;
                        break;
                    }
                }
            }
        }
        return stack.isEmpty() && answer;
    }
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.cpp
#include<string>
#include <iostream>

using namespace std;

bool solution(string s)
{
    bool answer = true;
    int count = 0;
    
    for(int i = 0; i < s.length() && count >= 0; ++i)
    {
        if (s[i] == '(')
            ++count;
        else
            --count;
    }
    
    if (count != 0)
        answer = false;

    return answer;
}